explainer-beforematch.md

```diff
@@ -0,0 +1,135 @@
+# The `beforematch` event
+
+### Summary (TL;DR)
+
+The `beforematch` event is fired at render timing in the following situations:
+
+* There is a new find-in-page active match, in which case the event is fired
+  with the matched element as the target.
+* There is a fragment-link navigation, in which case the event is fired with
+  the matched fragment as the target.
+* There is a scroll-to-text navigation, in which case the event is fired with
+  the matched element as the target.
+
```

Comment:
The use case for this event is to allow developers to help users find hidden content on the page. Hidden content includes collapsed sections, `visibility: hidden` subtrees, and `subtree-visibility: hidden-matchable` [link to spec] subtrees.

Reply:
We don't yet have a spec for hidden-matchable, but I'll add that.